controller/k8s/watcher.go

```diff
@@ -36,6 +41,18 @@ func (w *watcher) run() error {
 	initialized := make(chan struct{}, 1)
 	defer close(initialized)
 
+	if w.watchInit != nil {
```

Comment:
It will simplify the code if we assume that `watchInit` is always provided and never nil. Let's just remove this nil check.